pkg/controller/node/nodecontroller_test.go

```diff
@@ -458,191 +457,689 @@ func TestMonitorNodeStatusEvictPods(t *testing.T) {
+							// Node status was updated by nodecontroller 1hr ago
+							LastHeartbeatTime:  unversioned.Date(2015, 1, 1, 12, 0, 0, 0, time.UTC),
+							LastTransitionTime: unversioned.Date(2015, 1, 1, 12, 0, 0, 0, time.UTC),
+						},
 					},
 				},
 			},
+			expectedInitialStates: map[string]zoneState{
+				createZoneID("region1", "zone1"): stateFullDisruption,
+				createZoneID("region2", "zone2"): stateFullDisruption,
+			},
+			expectedFollowingStates: map[string]zoneState{
+				createZoneID("region1", "zone1"): stateFullDisruption,
+				createZoneID("region2", "zone2"): stateFullDisruption,
+			},
 			expectedEvictPods: false,
-			description:       "Network Segmentation: Node created long time ago, node controller posted Unknown for a long period of time on both Nodes.",
+			description:       "Network Disruption: Both zones down - eviction shouldn't take place.",
 		},
-		// NetworkSegmentation: Node created long time ago, node controller posted Unknown for a long period
-		// of on first Node, eviction should stop even though -master Node is healthy.
+		// NetworkDisruption: Node created long time ago, node controller posted Unknown for a long period of time on both Nodes.
+		// One zone is down - eviction should take place
 		{
-			fakeNodeHandler: &FakeNodeHandler{
-				Existing: []*api.Node{
-					{
-						ObjectMeta: api.ObjectMeta{
-							Name:              "node0",
-							CreationTimestamp: unversioned.Date(2012, 1, 1, 0, 0, 0, 0, time.UTC),
-							Labels: map[string]string{
-								unversioned.LabelZoneRegion:        "region1",
-								unversioned.LabelZoneFailureDomain: "zone1",
+			nodeList: []*api.Node{
+				{
+					ObjectMeta: api.ObjectMeta{
+						Name:              "node0",
+						CreationTimestamp: unversioned.Date(2012, 1, 1, 0, 0, 0, 0, time.UTC),
+						Labels: map[string]string{
+							unversioned.LabelZoneRegion:        "region1",
+							unversioned.LabelZoneFailureDomain: "zone1",
+						},
+					},
+					Status: api.NodeStatus{
+						Conditions: []api.NodeCondition{
+							{
+								Type:               api.NodeReady,
+								Status:             api.ConditionUnknown,
+								LastHeartbeatTime:  unversioned.Date(2015, 1, 1, 12, 0, 0, 0, time.UTC),
+								LastTransitionTime: unversioned.Date(2015, 1, 1, 12, 0, 0, 0, time.UTC),
 							},
 						},
-						Status: api.NodeStatus{
-							Conditions: []api.NodeCondition{
-								{
-									Type:               api.NodeReady,
-									Status:             api.ConditionUnknown,
-									LastHeartbeatTime:  unversioned.Date(2015, 1, 1, 12, 0, 0, 0, time.UTC),
-									LastTransitionTime: unversioned.Date(2015, 1, 1, 12, 0, 0, 0, time.UTC),
-								},
+					},
+				},
+				{
+					ObjectMeta: api.ObjectMeta{
+						Name:              "node1",
+						CreationTimestamp: unversioned.Date(2012, 1, 1, 0, 0, 0, 0, time.UTC),
+						Labels: map[string]string{
+							unversioned.LabelZoneRegion:        "region2",
+							unversioned.LabelZoneFailureDomain: "zone2",
```

Comment:
Are we using both values to detect zone? Maybe we should leave "region1" and only change "zone1" to "zone2" in this case?